test/test_train_imagenet.py

```diff
@@ -164,10 +166,13 @@ def test_loop_fn(model, loader, device, context):
     return correct / total_samples
 
   accuracy = 0.0
+  writer = SummaryWriter(log_dir=FLAGS.logdir)
```

Comment:
This needs to be created only if logdir is not None ... unless SummaryWriter is a noop if logdir is None.

Reply:
I tried running this without specifying --logdir and verified that None was being passed into SummaryWriter. When this happens, SummaryWriter works without complaining and defaults to creating a new directory: "runs/"

Reply:
No, we do not want that :wink: 
So lets create a write only if the user wants to (logdir not None).

Reply:
If writer is None when logdir is None, we will need to perform a check for `if writer:` at every spot where we call `writer.add_scalar`. Not a huge deal but adds to clutter

Reply:
That's OK. Dumping stuff into a random location is worse :wink:

Reply:
Added a helper method so it'll be a 1-line call whenever we write a summary data point. Verified that nothing gets written with `--logdir=None` and that valid summaries get written with `--logdir=non-None`